LaTeX/LaTeX.sublime-syntax

```diff
@@ -628,6 +629,36 @@ contexts:
           pop: true
         - include: main
 
+  footnote:
+    - match: |-
+        (?x)
+        ((\\)footnote(?:text)?)\s*
+        ((\[)
+            \s*\d*\s*
```

Comment:
The same applies to here. :-)

Reply:
Not sure how to change it without making it look more complicated

Like this?

```yaml
    - match: ((\\)footnote(?:text)?)\b
      captures:
        1: support.function.footnote.latex
        2: punctuation.definition.backslash.latex
      push:
        - meta_scope: meta.function.footnote.latex
        - include: general-optional-arguments
        - match: '\{'
          scope: punctuation.definition.group.brace.begin.latex
          set:
            - meta_scope: meta.function.footnote.latex meta.group.brace.latex
            - meta_content_scope: markup.italic.footnote.latex
            - match: '\}'
              scope: punctuation.definition.group.brace.end.latex
              pop: true
            - include: main
        - match: '(?=\S)'
          pop: true
```
or like this?
```yaml
    - match: ((\\)footnote(?:text)?)\b
      captures:
        1: support.function.footnote.latex
        2: punctuation.definition.backslash.latex
      push:
        - [{meta_scope: meta.function.footnote.latex}, {match: '', pop: true}]
        - - match: '\{'
            scope: meta.group.brace.latex punctuation.definition.group.brace.begin.latex
            set:
            - meta_content_scope: meta.group.brace.latex markup.italic.footnote.latex
            - match: '\}'
              scope: meta.group.brace.latex punctuation.definition.group.brace.end.latex
              pop: true
            - include: main
        - optional-arguments
```

Reply:
I personally find that the first one is more readable. Actually, the usage of `[{meta_scope: meta.function.footnote.latex}, {match: '', pop: true}]` was introduced in this [PR](https://github.com/sublimehq/Packages/commit/04064c47e6fbc579c93ac6293a3b3a78ea0d34c1). And TBH, I don't actually fully understand how it works.

Reply:
Wait, does the first one match the second brace in `\footnote{}{}`

Reply:
The first one would also match `\footnote[][]{}`, but it's imo not relevant whether it's matched or not.